ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -71,12 +93,15 @@ def __init__(self):
         self._current_activity = None
         self._next_activity = None
 
+        # Tracks whether or not a timer is currently active
+        self._timer_active = False
```

Comment:
This variable, since it's used accessed/set on the main and child threads, should have an associated lock: e.g. `self._timer_lock = threading.RLock()`.